File: Iteration2/code/CSE3063GRP12/src/core/models/concretes/NotificationResponse.java
package core.models.concretes;

import java.util.ArrayList;

public class NotificationResponse {
    private ArrayList<Notification> listOfNotification;
    private boolean containsNew;

    public NotificationResponse(ArrayList<Notification> listOfNotification, boolean containsNew) {
        this.listOfNotification = listOfNotification;
        this.containsNew = containsNew;
    }

    public void setListOfNotification(ArrayList<Notification> listOfNotification) {
        this.listOfNotification = listOfNotification;
    }

    public ArrayList<Notification> getListOfNotification() {
        return listOfNotification;
    }

    public boolean isContainsNew() {
        return containsNew;
    }

    public void setContainsNew(boolean containsNew) {
        this.containsNew = containsNew;
    }

}

File: Iteration3/code/CSE3063GRP12/src/core/models/concretes/NotificationResponse.py
class NotificationResponse:
    def __init__(self, listOfNotification, containsNew):
        self.listOfNotification = listOfNotification  # list of Notification instances
        self.containsNew = containsNew
    
    def to_dict(self):
        return {
            "listOfNotification": [notification.to_dict() if hasattr(notification, 'to_dict') else notification for notification in self.listOfNotification],
            "containsNew": self.containsNew
        }

